src/context.ts

```diff
@@ -2,9 +2,53 @@ import { arrayEqualsShallow, hashUnit, isObject, isPromise } from "./utils";
 import { VariantAssigner } from "./assigner";
 import { AudienceMatcher } from "./matcher";
 import { insertUniqueSorted } from "./algorithm";
+import SDK from "./sdk";
+import { ContextPublisher } from "./publisher";
+import { ContextDataProvider } from "./provider";
+import {
+	Attribute,
+	ContextData,
+	ContextOptions,
+	ContextParams,
+	EventLogger,
+	EventName,
+	Experiment,
+	ExperimentData,
+	Exposure,
+	Goal,
+	PublishParams,
+	Units,
+} from "./types";
+import { ClientRequestOptions } from "./types";
 
 export default class Context {
-	constructor(sdk, options, params, promise) {
+	private readonly _assigners: Record<string, VariantAssigner>;
+	private readonly _attrs: Attribute[];
+	private readonly _audienceMatcher: AudienceMatcher;
+	private readonly _cassignments: Record<string, number>;
+	private readonly _dataProvider: ContextDataProvider;
+	private readonly _eventLogger: EventLogger;
+	private readonly _opts: ContextOptions;
+	private readonly _publisher: ContextPublisher;
+	private readonly _sdk: SDK;
+	private readonly _units: Units;
+	private _assignments: Record<string, ExperimentData>;
```

Comment:
There should be a new type `Assignment` used here. See comments below.

Reply:
Done :)